bin/wabur

```diff
@@ -12,38 +12,45 @@ require 'logger'
 require 'wab'
 require 'wab/impl'
 
-# Initial options
-options = {
-  'base'        => '.',
-  'config_file' => nil
-}
-verbose_cnt = 0
+# The options for this application are determined by the Configuration class
+# which takes a usage string and an options map. The returned object is a
+# WAB::Impl::Configuration object that is used by the shell to configure
+# itself before being started.
 
-opts = OptionParser.new(%{Usage: wabur [options]
+usage = %{
+Usage: wabur [options]
 
-A WAB Runner that is configured with a UNIX style config file.
-})
-opts.on('-b', '--base PATH', String, 'App root directory. Defaults to "."') { |d| options['base'] = d }
-opts.on('-c', '--config PATH', String, 'Custom configuration file')         { |c| options['config_file'] = c }
-opts.on('-v', '--verbose', 'Verbose output')                                { verbose_cnt += 1 }
-opts.on('-h', '--help', 'Show this display')                                { puts opts.help; Process.exit!(0) }
-opts.parse(ARGV)
+A pure Ruby WAB Runner. Configuration is either by command line or a
+configuration file which can be a UNIX style conf file, a JSON file, or a YAML
+file. The configuration typically includes designating Controller classes for
+each URL path to be handled.
 
-config = WAB::Impl::Configuration.from(options)
+}
 
-if 0 < verbose_cnt
-  v = config['verbose'].to_i - verbose_cnt
-  v = 0 if v < 0
-  config['verbose'] = v
-end
+options = {
+  base: { val: '.', doc: 'App root directory which is $BASE.', short: '-b' },
+  store: {
+    dir: { val: File.join('$BASE', 'data'), doc: "Directory to use for data storage." }
+  },
+  path_prefix: { val: '/v1', doc: 'URL path prefix for relative handler routing.' },
+  handler: { val: [], doc: 'Type and handler/controller class in the form <type>=<controller class>.', short: '-t', parse: [ :type, :handler ] },
+  type_key: { val: 'kind', doc: 'Shell type_key.' },
+  http: {
+    dir: { val: File.join('$BASE', 'pages'), doc: 'Directory where HTTP content resides.' },
+    port: { val: 6363, doc: 'HTTP Port to listen on.' },
+  },
+  verbosity: { val: 'WARN', doc: 'Log level. (ERROR, WARN, INFO, DEBUG)' },
+}
```

Comment:
this whole section has now become visually unpleasing.. therefore IMO, visual noise